Hazel/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.16)
+project(Hazel)
+
+set(CMAKE_CXX_STANDARD 17)
+
+add_compile_definitions(HZ_DEBUG)
+
+include_directories(vendor/spdlog/include)
+include_directories(vendor/Glad/include)
+include_directories(vendor/GLFW/include)
+include_directories(vendor/imgui)
+include_directories(vendor/stb_image)
+include_directories(vendor/glm)
+
+add_subdirectory(vendor/glm)
+add_subdirectory(vendor/GLFW)
+
+set(IMGUI_SRC
+        vendor/imgui/imgui.h
+        vendor/imgui/imgui.cpp
+        vendor/imgui/imgui_widgets.cpp
+        vendor/imgui/imgui_demo.cpp
+        vendor/imgui/imgui_draw.cpp)
+
+set(STBIMG_SRC
+        vendor/stb_image/stb_image.cpp)
+
+set(GLAD_SRC
+        vendor/Glad/src/glad.c)
+
+set(SOURCES
+        src/Hazel/Core/Application.cpp
```

Comment:
I think it's better if you just glob the files here instead of hardcoding all the source files to automate the process,

file(GLOB SRC_FILES "src/\*.c\* src/\*.c/\*.c\* src/\*.c/\*.c\*/\*.c\* src/\*.h\* src/\*.h/\*.h\* src/\*.h/\*.h\*/\*.h\*)

Then just use $SRC_FILES as necessary

Reply:
you can also just `file(GLOB_RECURSE SRC_FILES src/*)`